src/spikeinterface/core/core_tools.py

```diff
@@ -851,6 +852,116 @@ def recursive_path_modifier(d, func, target="path", copy=True) -> dict:
                     raise ValueError(f"{k} key for path  must be str or list[str]")
 
 
+
+def _get_paths_list(d):
+    # this explore a dict and get all paths flatten in a list
+    # the trick is to use a closure func called by recursive_path_modifier()
+    path_list = []
+    def append_to_path(p):
+        path_list.append(p)
+    recursive_path_modifier(d, append_to_path, target="path", copy=True)
+    return path_list
+
+
+def _relative_to(p, relative_folder):
+    # custum os.path.relpath() with more checks
+
+    relative_folder = Path(relative_folder).resolve().absolute()
```

Comment:
```suggestion
    relative_folder = Path(relative_folder).resolve()
```
Resolve does what absolute does:
https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve